tests/bsim/bluetooth/mesh/src/mesh_test.h

```diff
@@ -23,6 +23,8 @@
 
 #include <zephyr/bluetooth/bluetooth.h>
 #include <zephyr/bluetooth/mesh.h>
+#include <zephyr/bluetooth/hci.h>
```

Comment:
Why do you need hci header?

Reply:
Used in `bt_mesh_test_wait_for_adv` in `mesh_test.c`.

```
...
  struct bt_le_scan_param scan_param = {
    .type       = BT_HCI_LE_SCAN_PASSIVE,
    .options    = BT_LE_SCAN_OPT_NONE,
    .interval   = BT_MESH_ADV_SCAN_UNIT(1000),
    .window     = BT_MESH_ADV_SCAN_UNIT(1000)
  };
```

Reply:
If this is required only for provided above code why it is here instead of c file?